### DISCUSSION TITLE
DeclarativeBase is unknown import symbol

### DISCUSSION THREAD

**Discussion by c0dezer019** [sqlalchemy/sqlalchemy#10702] [Usage Questions]
### Ensure stubs packages are not installed

- [X] No sqlalchemy stub packages is installed (both `sqlalchemy-stubs` and `sqlalchemy2-stubs` are not compatible with v2)

### Verify if the api is typed

- [X] The api is not in a module listed in [#6810](https://github.com/sqlalchemy/sqlalchemy/issues/6810) so it should pass type checking

### Describe the typing issue

Pylance reports that DeclarativeBase is an unknown import symbol. It is typed as Unknown. It was working yesterday and now it isn't. I didn't do anything to change the env that would cause it to do this.

### To Reproduce

```python
Using a pipenv venv, install sqlalchemy.

`from sqlalchemy.orm import DeclarativeBase`
```

### Error

```
"DeclarativeBase" is unknown import symbol Pylance (reportGeneralTypeIssues)
``` 

### Versions

- OS: Lubuntu
- Python: 3.10.12
- SQLAlchemy: latest
- Type checker (eg: mypy 0.991, pyright 1.1.290, etc): Pylance 2023-11-15, Pyright

**Comment by zzzeek** [comment]
hi -

it's most likely that the wrong version of SQLAlchemy, like a 1.4 version, is installed.  

confirm which SQLAlchemy you have like this:

```py

# sqlalchemy 2.0.x
$ python
Python 3.12.0 (main, Oct  2 2023, 00:00:00) [GCC 13.2.1 20230918 (Red Hat 13.2.1-3)] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> from sqlalchemy.orm import DeclarativeBase
>>> import sqlalchemy
>>> sqlalchemy.__version__
'2.0.23'




# sqlalchemy 1.4.x

$ python
Python 3.12.0 (main, Oct  2 2023, 00:00:00) [GCC 13.2.1 20230918 (Red Hat 13.2.1-3)] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> from sqlalchemy.orm import DeclarativeBase
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
ImportError: cannot import name 'DeclarativeBase' from 'sqlalchemy.orm' (/home/classic/dev/sqlalchemy/lib/sqlalchemy/orm/__init__.py). Did you mean: 'declarative_base'?
>>> import sqlalchemy
>>> sqlalchemy.__version__
'1.4.50'

```

**Reply by zzzeek** [reply]
as far as pylance here's the latest vscode/pylance

![image](https://github.com/sqlalchemy/sqlalchemy/assets/128223/18206f38-2d8e-4fbc-b199-233e1c1eae06)

**Comment by c0dezer019** [comment]
It could be a versioning issue, but it isn't sqlalchemy as I have the latest. Really wish I could figure this out, because now it's working again.

**Comment by c0dezer019** [comment]
Ok so, apparently for whatever reason it dropped down to 1.4.5, ran `pipenv update` and it re-updated to 2.0.23. Even though in my Pipfile I have it set to `sqlalchemy = "*"` so I dunno. I set it to explicitly use the latest version or greater so hopefully it won't happen again.

**Comment by MikeWent** [comment]
In my case, `sqlalchemy-stubs` was an issue. Probably it doesn't work with sqlalchemy 3, breaking pylance.

`pipenv uninstall sqlalchemy-stubs` did a thing,

**Reply by dryleaf** [reply]
This fixed the issue for me. Thanks
